internal/app/components/courseSelections/courseSelections.go

```diff
@@ -24,34 +24,50 @@ func NewCourseSelection(name CourseSelectionName) *CourseSelection {
 	return &CourseSelection{
 		Id:             newCourseSelectionId(),
 		Name:           name,
-		TermSelections: []*courses.TermSelection{},
-		Plans:          []plans.Plan{},
+		TermSelections: []*TermSelection{},
+		Plans:          plans.Plans{},
 	}
 }
 
-func (cs CourseSelection) AddTermSelection(ts *courses.TermSelection) {
+func (cs CourseSelection) AddTermSelection(ts *TermSelection) {
 	cs.TermSelections = append(cs.TermSelections, ts)
 }
 
 // Flattens TermSelections and aggregates it into one CourseRecords
 func (cs CourseSelection) Aggregate() *courses.CourseRecords {
 	records := courses.CourseRecords{}
 	for _, ts := range cs.TermSelections {
-		for id, record := range ts.CourseRecords {
-			records[id] = record
+		for _, record := range ts.CourseRecords {
+			records = append(records, record)
 		}
 	}
 	return &records
 }
 
-// TODO: make it return the plans that are not satisfied
 // Checks if all declared plans in Plans are satisfied
-func (cs CourseSelection) IsSatisfied() bool {
+func (cs CourseSelection) IncompletePlans() plans.Plans {
 	records := cs.Aggregate()
+	var incompletePlans plans.Plans
 	for _, plan := range cs.Plans {
-		if !plan.IsCompleted(records) {
-			return false
+		if !isPlanSatisfied(plan, records) {
+			incompletePlans = append(incompletePlans, plan)
 		}
 	}
-	return true
+	return incompletePlans
+}
+
+func isPlanSatisfied(plan *plans.Plan, records *courses.CourseRecords) bool {
```

Comment:
what's the point of this function? why not just call Plan::IsCompleted directly?

Reply:
true, i think i was making some other function but it ended up like this LOL